core-spi-test/src/main/java/org/ehcache/internal/store/StoreFactory.java

```diff
@@ -51,6 +52,8 @@
 
   ServiceConfiguration<?>[] getServiceConfigurations();
 
+  ServiceProvider getServiceProvider();
```

Comment:
I'm not entirely convinced by your arguments but as you said, this is rather minor. The StoreFactory already requires some cleanups, let's leave that one for that time too.